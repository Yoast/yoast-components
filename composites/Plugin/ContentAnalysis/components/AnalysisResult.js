import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import colors from "../../../../style-guide/colors.json";

import { circle } from "../../../../style-guide/svg/index";
import { eye } from "../../../../style-guide/svg/index";
import { Icon } from "../../Shared/components/Icon.js";
import IconButtonToggle from "../../Shared/components/IconButtonToggle.js";

/**
 * Returns a AnalysisResultBase component.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} The rendered AnalysisResultBase component.
 */
const AnalysisResultBase = styled.li`
	// This is the height of the IconButtonToggle.
	min-height: 24px;
	padding: 4px 4px;
	display: inline-flex;
	align-items: center;
`;

const AnalysisResultText = styled.p`
	margin: 0 8px;
`;

export const AnalysisResult = ( props ) => {
	return (
		<AnalysisResultBase { ...props }>
			<Icon
				icon={ circle }
				color={ props.bulletColor }
				size="13px"
				aria-label={ "SEO score " + props.score }
				aria-hidden={ false }
				focusable={ true }
			/>
			<AnalysisResultText>{ props.resultText }</AnalysisResultText>
			{
				props.eyeButtonIsActive
					? <IconButtonToggle
						onClick={ props.onClick }
						id={ props.id }
						icon={ eye }
						pressed={ props.pressed }
						ariaLabel={ props.ariaLabel }
					/>
					: null
			}
		</AnalysisResultBase>
	);
};

AnalysisResult.propTypes = {
	resultText: PropTypes.string.isRequired,
	bulletColor: PropTypes.string.isRequired,
	eyeButtonIsActive: PropTypes.bool.isRequired,
	id: PropTypes.string.isRequired,
	pressed: PropTypes.bool.isRequired,
	ariaLabel: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired,
	score: PropTypes.string.isRequired,
};

export default AnalysisResult;
