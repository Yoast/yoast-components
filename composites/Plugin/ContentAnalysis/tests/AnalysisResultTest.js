import React from "react";
import { createComponentWithIntl } from "../../../../utils/intlProvider";
import AnalysisResult from "../components/AnalysisResult.js";

test( "the AnalysisResult component matches the snapshot", () => {
	const component = createComponentWithIntl(
		<AnalysisResult
			ariaLabel="SEOResult"
			bulletColor="blue"
			buttonId="Result button"
			pressed={ true }
			eyeButtonIsActive={ true }
			onButtonClick={ () => {} }
			resultText={ "You're doing great!" }
			score="good"
		/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
