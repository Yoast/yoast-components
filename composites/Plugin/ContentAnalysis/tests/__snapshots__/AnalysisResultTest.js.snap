// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the AnalysisResult component matches the snapshot 1`] = `
<li
  className="sc-bwzfXH gqJfBa"
>
  <span
    aria-label="SEO score good"
    role="img"
  >
    <svg
      aria-hidden="true"
      className="sc-bxivhb hpJpJA"
      focusable="false"
      role="presentation"
    />
  </span>
  <p
    className="sc-htpNat fgCHJU"
  >
    You're doing great!
  </p>
  <button
    aria-label="SEOResult"
    aria-pressed={true}
    className="sc-bdVaJa dlGzMO"
    id="Result button"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="sc-ifAKCX fzBsVd"
      focusable="false"
      role="img"
    />
  </button>
</li>
`;
